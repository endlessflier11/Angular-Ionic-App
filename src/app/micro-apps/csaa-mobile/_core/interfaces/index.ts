export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
