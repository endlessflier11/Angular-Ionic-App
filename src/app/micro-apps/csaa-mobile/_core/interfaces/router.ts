import { Route, UrlTree } from '@angular/router';
import { NavigationOptions } from '@ionic/angular/providers/nav-controller';
import { Observable } from 'rxjs';

interface AppRoute extends Route {
  name?: string;
  isGroupIndex?: boolean; // If true, it means route is the index of parent. Then it inherits all parent properties
}

export type AppRoutes = AppRoute[];

export interface RouteParam {
  [key: string]: string;
}

export interface NamedRouteConfig {
  name: string;
  path: string;
  children?: NamedRouteConfig[];
}

export interface NamedRoute {
  name: string;
  path: string;
  relativePath: string;
}

export interface CanComponentDeactivate {
  canDeactivate: () =>
    | Observable<boolean | UrlTree>
    | Promise<boolean | UrlTree>
    | boolean
    | UrlTree;
}

export interface CsaaRouteStackItem {
  name: string;
  params?: RouteParam;
  options?: NavigationOptions;
}
