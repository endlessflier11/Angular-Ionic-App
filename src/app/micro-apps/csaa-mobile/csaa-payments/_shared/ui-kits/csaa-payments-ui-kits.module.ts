import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
