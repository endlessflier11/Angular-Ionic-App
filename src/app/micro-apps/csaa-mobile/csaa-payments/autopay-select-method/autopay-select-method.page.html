<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot="start">
      <csaa-back-button (back)="onClickBackBtn()" text="Cancel"></csaa-back-button>
    </ion-buttons>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <ng-container *ngIf="wallet?.paymentAccounts.length > 0">
    <h2 class="ion-text-center">Wallet</h2>
    <ion-list>
      <ng-container
        *ngFor="let paymentAccount of wallet.paymentAccounts; let i = index, let isLast = last;"
      >
        <ion-item
          class="wallet-item item"
          (click)="selectPaymentAccount(paymentAccount)"
          [lines]="isLast ? 'full' : undefined"
        >
          <ion-avatar slot="start">
            <span><img [src]="generateIconSrc(paymentAccount)" [alt]="undefined" /></span>
          </ion-avatar>
          <ion-label>
            <h3>{{ paymentAccount.shortName }}</h3>
            <ng-container *ngIf="paymentAccount.card">
              <p [class]="{'card-expired': paymentAccount.card.expired}">
                {{ paymentAccount.card.expired ? 'Expired' : paymentAccount.card.last4digits +' '+
                (paymentAccount.card.type | titlecase) }}
              </p>
            </ng-container>
            <ng-container *ngIf="paymentAccount.account">
              <p>
                {{ paymentAccount.account.accountNumber }} {{ paymentAccount.account.bankAccountType
                }}
              </p>
            </ng-container>
          </ion-label>
          <ion-button
            slot="end"
            fill="none"
            (click)="onClickEditPaymentAccount(paymentAccount); $event.stopPropagation()"
          >
            EDIT
          </ion-button>
        </ion-item>
      </ng-container>
    </ion-list>
  </ng-container>

  <h2 class="ion-text-center">Payment Method</h2>
  <ion-list class="ion-margin-bottom">
    <ion-item
      *ngFor="let item of newPaymentAccountMenu, let isLast = last"
      [lines]="isLast ? 'none' : 'full'"
      detail
      [routerLink]="route(item.route)"
      [queryParams]="{ autopay: true, policyNumber: policyNumber }"
      >{{ item.title}}</ion-item
    >
  </ion-list>

  <ng-container *ngIf="autopayEnrolledPayments.length">
    <h2 class="ion-text-center">AutoPay</h2>

    <ion-grid class="ion-no-padding">
      <ng-container *ngFor="let payment of autopayEnrolledPayments">
        <ion-row class="ion-padding-top ion-padding-bottom" (click)="goToAutopay($event,payment)">
          <ion-col>
            <ion-item lines="none" detail>
              <div class="policy-meta-content">
                <h3>{{ payment.policyNumber }}</h3>
                <span class="subtitle">{{ autopayLabel(payment.autopayEnrollment) }}</span>
              </div>
              <div slot="end" class="policy-meta-icon">
                <csaa-theme-icon
                  slot="end"
                  [type]="payment.policyType"
                  [size]="40"
                ></csaa-theme-icon>
              </div>
            </ion-item>
            <ion-item lines="none">
              <csaa-policy-subtitles [policySubtitle]="payment.subtitle"></csaa-policy-subtitles>
            </ion-item>
          </ion-col>
        </ion-row>
      </ng-container>
    </ion-grid>
  </ng-container>
</ion-content>
