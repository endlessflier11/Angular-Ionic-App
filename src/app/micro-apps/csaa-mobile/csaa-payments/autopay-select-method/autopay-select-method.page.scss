@import '../../_scss/variables';

ion-header {
  ion-toolbar {
    --border-width: 0;
  }
}
ion-content {
  --background: var(--ion-color-primary-contrast);
  --padding-start: 0;
  --padding-end: 0;

  h2 {
    font-weight: 600;
    font-size: 24px;
  }

  ion-list {
    margin-bottom: 32px;

    ion-item {
      --detail-icon-opacity: 1;

      &.wallet-item {
        &:first-child {
          border-top: 0.55px solid #c8c7cc;
        }

        ion-avatar {
          --border-radius: 0;

          > span {
            display: flex;
            height: 100%;
          }
        }

        h3 {
          font-weight: 400;
          font-size: 17px;
          white-space: nowrap;
          margin: 0;
        }

        p {
          color: $csaa-lighter-color;

          &.card-expired {
            color: $csaa-emergency-color;
          }
        }

        ion-button {
          font-size: 15px;
          font-weight: 400;
          color: var(--ion-color-primary);
        }
      }
    }
  }

  ion-grid {
    ion-row {
      ion-col {
        ion-item {
          --detail-icon-color: var(--ion-color-primary);
          --detail-icon-opacity: 1;

          &:nth-child(1) {
            margin-bottom: 4px;
          }
          &:nth-child(2) {
            --min-height: 0;
          }

          h3 {
            margin-top: 0;
            margin-bottom: 0;
          }

          span {
            font-size: 14px;
            color: var(--ion-color-medium);
          }
        }
      }
    }
  }
}
