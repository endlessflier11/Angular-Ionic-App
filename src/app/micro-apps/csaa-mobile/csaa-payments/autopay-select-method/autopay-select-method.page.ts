import { Component, OnDestroy, OnInit } from '@angular/core';
import { noop, Subject } from 'rxjs';
import { WalletDetails, PaymentAccount, UpcomingPayment, Policy } from '../../_core/interfaces';
import { ConfigService, RouterService } from '../../_core/services';
import { MakePaymentService } from '../_shared/services/make-payment.service';
import { PaymentHelper } from '../../_core/shared/payment.helper';
import { ActivatedRoute } from '@angular/router';
import { PaymentState } from '../../_core/store/states/payment.state';
import { Store } from '@ngxs/store';
import { PolicyState } from '../../_core/store/states/policy.state';

const newPaymentAccountMenu: { route: string; title: string }[] = [
  { route: 'csaa.payment.method.card', title: 'Credit or Debit Card' },
  { route: 'csaa.payment.method.checking-account', title: 'Checking Account' },
  { route: 'csaa.payment.method.savings-account', title: 'Savings Account' },
];

@Component({
  selector: 'app-csaa-payment-select-method',
  templateUrl: './autopay-select-method.page.html',
  styleUrls: ['./autopay-select-method.page.scss'],
})
export class AutopaySelectMethodPage implements OnInit, OnDestroy {
  private readonly tearDown$: Subject<any> = new Subject();
  public readonly autopayLabel = PaymentHelper.getAutopayEnrollmentLabel;
  public readonly newPaymentAccountMenu = newPaymentAccountMenu;
  public currentTheme: string;
  public policyNumber: string;
  public policy: Policy;
  public wallet: WalletDetails;
  private upcomingPayments: UpcomingPayment[];

  public get autopayEnrolledPayments(): UpcomingPayment[] {
    return this.upcomingPayments ? this.upcomingPayments.filter((p) => p.autopayEnrollment) : [];
  }

  constructor(
    private readonly configService: ConfigService,
    private readonly routerService: RouterService,
    private readonly makePaymentService: MakePaymentService,
    private readonly activatedRoute: ActivatedRoute,
    private readonly store: Store
  ) {
    this.currentTheme = this.configService.getTheme();
  }

  ngOnInit() {
    this.activatedRoute.paramMap.subscribe((paramMap) => {
      this.policyNumber = paramMap.get('policyNumber');
      this.policy = this.store.selectSnapshot(PolicyState.policyData(this.policyNumber));
      this.wallet = this.store.selectSnapshot(PaymentState.walletDetails);
      this.upcomingPayments = this.store.selectSnapshot(PaymentState.upcomingPayments);
    });
  }

  ngOnDestroy() {
    this.tearDown$.next(true);
    this.tearDown$.complete();
  }

  route(path: string) {
    return this.routerService.fullPath(path);
  }

  onClickBackBtn() {
    this.navigateBack();
  }

  navigateBack(): void {
    this.routerService
      .navigateBack('csaa.payment.autopay.settings', { policyNumber: this.policyNumber })
      .then(noop);
  }

  goToAutopay($event, payment: UpcomingPayment) {
    $event.stopPropagation();
    $event.preventDefault();
    this.routerService
      .navigateBack('csaa.payment.autopay.settings', { policyNumber: payment.policyNumber })
      .then(noop);
  }

  generateIconSrc(account: PaymentAccount): string {
    const iconName: string = !account.card ? 'bank' : account.card.type;
    return `/assets/csaa-mobile/vectors/${iconName}.svg`;
  }

  selectPaymentAccount(account: PaymentAccount): void {
    this.makePaymentService.selectPaymentAccount(account);
    this.navigateBack();
  }

  onClickEditPaymentAccount(account: PaymentAccount): void {
    const normalizeBankType = (a: PaymentAccount): string =>
      a.account.bankAccountType.toLocaleLowerCase() === 'savings' ? 'savings' : 'checking';
    const suffix = account.card ? 'card' : normalizeBankType(account) + '-account';
    this.routerService
      .navigateForward(
        `csaa.payment.method.${suffix}.edit`,
        {
          account: account.paymentAccountToken,
        },
        { queryParams: { autopay: 'true', policyNumber: this.policyNumber } }
      )
      .then(noop);
  }
}
