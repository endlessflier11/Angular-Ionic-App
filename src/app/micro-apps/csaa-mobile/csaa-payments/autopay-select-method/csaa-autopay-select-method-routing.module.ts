import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AppRoutes } from '../../_core/interfaces';
import { AutopaySelectMethodPage } from './autopay-select-method.page';

const routes: AppRoutes = [
  {
    path: '',
    isGroupIndex: true,
    component: AutopaySelectMethodPage,
  },
];

@NgModule({
  declarations: [],
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class CsaaAutopaySelectMethodRoutingModule {}
