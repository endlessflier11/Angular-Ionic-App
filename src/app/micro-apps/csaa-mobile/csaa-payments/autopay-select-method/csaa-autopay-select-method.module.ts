import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CsaaAutopaySelectMethodRoutingModule } from './csaa-autopay-select-method-routing.module';
import { AutopaySelectMethodPage } from './autopay-select-method.page';
import { IonicModule } from '@ionic/angular';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentsUiKitsModule } from '../_shared/ui-kits/csaa-payments-ui-kits.module';

@NgModule({
  declarations: [AutopaySelectMethodPage],
  imports: [
    CommonModule,
    CsaaAutopaySelectMethodRoutingModule,
    IonicModule,
    UiKitsModule,
    CsaaPaymentsUiKitsModule,
  ],
})
export class CsaaAutopaySelectMethodModule {}
