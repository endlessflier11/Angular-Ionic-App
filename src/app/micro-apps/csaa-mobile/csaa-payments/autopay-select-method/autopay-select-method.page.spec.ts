/* eslint-disable @typescript-eslint/naming-convention,no-underscore-dangle */
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { By, CONFIG_STATE_FIXTURE_MOCK, PageTestingModule } from '@app/testing';
import { AutopaySelectMethodPage } from './autopay-select-method.page';
import { MakePaymentService } from '../_shared/services/make-payment.service';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentsUiKitsModule } from '../_shared/ui-kits/csaa-payments-ui-kits.module';
import { ActivatedRoute } from '@angular/router';
import { of } from 'rxjs';
import { CsaaAppInjector } from '../../csaa-app.injector';
import { Injector } from '@angular/core';
import { NgxsModule, Store } from '@ngxs/store';
import { AppState } from '../../../../_core/store';
import { Policy, PolicyStatus } from '../../_core/interfaces';

describe('AutopaySelectMethodPage', () => {
  let component: AutopaySelectMethodPage;
  let fixture: ComponentFixture<AutopaySelectMethodPage>;

  beforeEach(async(() => {
    const PARAM_MAP = new Map();
    PARAM_MAP.set('policyNumber', 'demo');
    TestBed.configureTestingModule({
      declarations: [AutopaySelectMethodPage],
      imports: [
        UiKitsModule,
        CsaaPaymentsUiKitsModule,
        PageTestingModule.withConfig({
          providesStorage: true,
          providesConfig: true,
          providesRouter: true,
        }),
        NgxsModule.forRoot([AppState]),
      ],
      providers: [
        MakePaymentService,
        { provide: ActivatedRoute, useValue: { paramMap: of(PARAM_MAP) } },
      ],
    }).compileComponents();
    const store = TestBed.inject(Store);
    store.reset({
      __csaa_app: {
        ...store.snapshot().__csaa_app,
        config: CONFIG_STATE_FIXTURE_MOCK,
      },
    });
    CsaaAppInjector.injector = TestBed.inject(Injector);
    fixture = TestBed.createComponent(AutopaySelectMethodPage);
    component = fixture.componentInstance;
    fixture.detectChanges();
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should create a snapshot', () => {
    const mockPolicy: Policy = {
      type: 2,
      id: '1',
      subtitle: '2018 Porsche 911',
      number: '1234',
      status: PolicyStatus.ACTIVE,
      riskState: 'AZ',
      productCode: '1',
      policyPrefix: 'AU_SS',
    };
    component.policy = mockPolicy;
    component.policyNumber = '1234567';
    component.wallet = {
      ownerId: '12345',
      walletId: '54321',
      paymentAccounts: [
        {
          account: {
            institution: {
              routingNumber: '123456',
            },
            accountNumber: '12345',
            holderName: 'Jane Doe',
            type: 'Checking',
            bankAccountType: 'Checking',
          },
          card: {
            holderName: 'Jane Doe',
            type: 'Debit',
            last4digits: '1111',
            printedName: 'Jane Doe',
            zipCode: '12345',
            expirationDate: '10/24',
            expired: false,
            isDebitCard: true,
          },
          isPreferred: true,
          paymentAccountToken: '1234567',
          shortName: 'test',
        },
      ],
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should display the Credit or Debit Card payment method', () => {
    const mockPolicy: Policy = {
      type: 2,
      id: '1',
      subtitle: '2018 Porsche 911',
      number: '1234',
      status: PolicyStatus.ACTIVE,
      riskState: 'AZ',
      productCode: '1',
      policyPrefix: 'AU_SS',
    };
    component.policy = mockPolicy;
    component.policyNumber = '1234567';
    component.wallet = {
      ownerId: '12345',
      walletId: '54321',
      paymentAccounts: [
        {
          account: {
            institution: {
              routingNumber: '123456',
            },
            accountNumber: '12345',
            holderName: 'Jane Doe',
            type: 'Checking',
            bankAccountType: 'Checking',
          },
          card: {
            holderName: 'Jane Doe',
            type: 'Debit',
            last4digits: '1111',
            printedName: 'Jane Doe',
            zipCode: '12345',
            expirationDate: '10/24',
            expired: false,
            isDebitCard: true,
          },
          isPreferred: true,
          paymentAccountToken: '1234567',
          shortName: 'test',
        },
      ],
    };
    const creditOrDebitCardButton = fixture.debugElement.query(
      By.cssAndText('ion-item', 'Credit or Debit Card')
    );
    expect(creditOrDebitCardButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should display the Checking Account method', () => {
    const mockPolicy: Policy = {
      type: 2,
      id: '1',
      subtitle: '2018 Porsche 911',
      number: '1234',
      status: PolicyStatus.ACTIVE,
      riskState: 'AZ',
      productCode: '1',
      policyPrefix: 'AU_SS',
    };
    component.policy = mockPolicy;
    component.policyNumber = '1234567';
    component.wallet = {
      ownerId: '12345',
      walletId: '54321',
      paymentAccounts: [
        {
          account: {
            institution: {
              routingNumber: '123456',
            },
            accountNumber: '12345',
            holderName: 'Jane Doe',
            type: 'Checking',
            bankAccountType: 'Checking',
          },
          card: {
            holderName: 'Jane Doe',
            type: 'Debit',
            last4digits: '1111',
            printedName: 'Jane Doe',
            zipCode: '12345',
            expirationDate: '10/24',
            expired: false,
            isDebitCard: true,
          },
          isPreferred: true,
          paymentAccountToken: '1234567',
          shortName: 'test',
        },
      ],
    };
    const checkingAccountButton = fixture.debugElement.query(
      By.cssAndText('ion-item', 'Checking Account')
    );
    expect(checkingAccountButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should display the Savings Account method', () => {
    const mockPolicy: Policy = {
      type: 2,
      id: '1',
      subtitle: '2018 Porsche 911',
      number: '1234',
      status: PolicyStatus.ACTIVE,
      riskState: 'AZ',
      productCode: '1',
      policyPrefix: 'AU_SS',
    };
    component.policy = mockPolicy;
    component.policyNumber = '1234567';
    component.wallet = {
      ownerId: '12345',
      walletId: '54321',
      paymentAccounts: [
        {
          account: {
            institution: {
              routingNumber: '123456',
            },
            accountNumber: '12345',
            holderName: 'Jane Doe',
            type: 'Checking',
            bankAccountType: 'Checking',
          },
          card: {
            holderName: 'Jane Doe',
            type: 'Debit',
            last4digits: '1111',
            printedName: 'Jane Doe',
            zipCode: '12345',
            expirationDate: '10/24',
            expired: false,
            isDebitCard: true,
          },
          isPreferred: true,
          paymentAccountToken: '1234567',
          shortName: 'test',
        },
      ],
    };
    const savingsAccountButton = fixture.debugElement.query(
      By.cssAndText('ion-item', 'Savings Account')
    );
    expect(savingsAccountButton).toBeTruthy();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should click on the cancel button', () => {
    const mockPolicy: Policy = {
      type: 2,
      id: '1',
      subtitle: '2018 Porsche 911',
      number: '1234',
      status: PolicyStatus.ACTIVE,
      riskState: 'AZ',
      productCode: '1',
      policyPrefix: 'AU_SS',
    };
    component.policy = mockPolicy;
    component.policyNumber = '1234567';
    component.wallet = {
      ownerId: '12345',
      walletId: '54321',
      paymentAccounts: [
        {
          account: {
            institution: {
              routingNumber: '123456',
            },
            accountNumber: '12345',
            holderName: 'Jane Doe',
            type: 'Checking',
            bankAccountType: 'Checking',
          },
          card: {
            holderName: 'Jane Doe',
            type: 'Debit',
            last4digits: '1111',
            printedName: 'Jane Doe',
            zipCode: '12345',
            expirationDate: '10/24',
            expired: false,
            isDebitCard: true,
          },
          isPreferred: true,
          paymentAccountToken: '1234567',
          shortName: 'test',
        },
      ],
    };
    const editButton = fixture.debugElement.query(By.cssAndText('ion-button', 'Cancel'));
    expect(editButton).toBeTruthy();
    editButton.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
