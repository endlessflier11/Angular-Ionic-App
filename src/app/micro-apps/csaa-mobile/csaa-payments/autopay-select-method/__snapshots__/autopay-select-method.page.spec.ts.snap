// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutopaySelectMethodPage should click on the cancel button 1`] = `
<app-csaa-payment-select-method
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  autopayLabel={[Function Function]}
  configService={[Function ConfigMockService]}
  currentTheme=""
  makePaymentService={[Function MakePaymentService]}
  newPaymentAccountMenu={[Function Array]}
  policy={[Function Object]}
  policyNumber={[Function String]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  tearDown$={[Function Subject]}
  upcomingPayments="undefined"
  wallet={[Function Object]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button
          text="Cancel"
        >
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Cancel
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <h2
      class="ion-text-center"
    >
      Wallet
    </h2>
    <ion-list>
      <ion-item
        class="wallet-item item"
      >
        <ion-avatar
          slot="start"
        >
          <span>
            <img
              alt="undefined"
              src="/assets/csaa-mobile/vectors/Debit.svg"
            />
          </span>
        </ion-avatar>
        <ion-label>
          <h3>
            test
          </h3>
          <p>
             1111 Debit 
          </p>
          <p>
             12345 Checking 
          </p>
        </ion-label>
        <ion-button
          fill="none"
          slot="end"
        >
           EDIT 
        </ion-button>
      </ion-item>
    </ion-list>
    <h2
      class="ion-text-center"
    >
      Payment Method
    </h2>
    <ion-list
      class="ion-margin-bottom"
    >
      <ion-item
        detail=""
        tabindex="0"
      >
        Credit or Debit Card
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Checking Account
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Savings Account
      </ion-item>
    </ion-list>
  </ion-content>
</app-csaa-payment-select-method>
`;

exports[`AutopaySelectMethodPage should create a snapshot 1`] = `
<app-csaa-payment-select-method
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  autopayLabel={[Function Function]}
  configService={[Function ConfigMockService]}
  currentTheme=""
  makePaymentService={[Function MakePaymentService]}
  newPaymentAccountMenu={[Function Array]}
  policy={[Function Object]}
  policyNumber={[Function String]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  tearDown$={[Function Subject]}
  upcomingPayments="undefined"
  wallet={[Function Object]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button
          text="Cancel"
        >
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Cancel
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <h2
      class="ion-text-center"
    >
      Wallet
    </h2>
    <ion-list>
      <ion-item
        class="wallet-item item"
      >
        <ion-avatar
          slot="start"
        >
          <span>
            <img
              alt="undefined"
              src="/assets/csaa-mobile/vectors/Debit.svg"
            />
          </span>
        </ion-avatar>
        <ion-label>
          <h3>
            test
          </h3>
          <p>
             1111 Debit 
          </p>
          <p>
             12345 Checking 
          </p>
        </ion-label>
        <ion-button
          fill="none"
          slot="end"
        >
           EDIT 
        </ion-button>
      </ion-item>
    </ion-list>
    <h2
      class="ion-text-center"
    >
      Payment Method
    </h2>
    <ion-list
      class="ion-margin-bottom"
    >
      <ion-item
        detail=""
        tabindex="0"
      >
        Credit or Debit Card
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Checking Account
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Savings Account
      </ion-item>
    </ion-list>
  </ion-content>
</app-csaa-payment-select-method>
`;

exports[`AutopaySelectMethodPage should display the Checking Account method 1`] = `
<app-csaa-payment-select-method
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  autopayLabel={[Function Function]}
  configService={[Function ConfigMockService]}
  currentTheme=""
  makePaymentService={[Function MakePaymentService]}
  newPaymentAccountMenu={[Function Array]}
  policy={[Function Object]}
  policyNumber={[Function String]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  tearDown$={[Function Subject]}
  upcomingPayments="undefined"
  wallet={[Function Object]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button
          text="Cancel"
        >
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Cancel
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <h2
      class="ion-text-center"
    >
      Wallet
    </h2>
    <ion-list>
      <ion-item
        class="wallet-item item"
      >
        <ion-avatar
          slot="start"
        >
          <span>
            <img
              alt="undefined"
              src="/assets/csaa-mobile/vectors/Debit.svg"
            />
          </span>
        </ion-avatar>
        <ion-label>
          <h3>
            test
          </h3>
          <p>
             1111 Debit 
          </p>
          <p>
             12345 Checking 
          </p>
        </ion-label>
        <ion-button
          fill="none"
          slot="end"
        >
           EDIT 
        </ion-button>
      </ion-item>
    </ion-list>
    <h2
      class="ion-text-center"
    >
      Payment Method
    </h2>
    <ion-list
      class="ion-margin-bottom"
    >
      <ion-item
        detail=""
        tabindex="0"
      >
        Credit or Debit Card
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Checking Account
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Savings Account
      </ion-item>
    </ion-list>
  </ion-content>
</app-csaa-payment-select-method>
`;

exports[`AutopaySelectMethodPage should display the Credit or Debit Card payment method 1`] = `
<app-csaa-payment-select-method
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  autopayLabel={[Function Function]}
  configService={[Function ConfigMockService]}
  currentTheme=""
  makePaymentService={[Function MakePaymentService]}
  newPaymentAccountMenu={[Function Array]}
  policy={[Function Object]}
  policyNumber={[Function String]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  tearDown$={[Function Subject]}
  upcomingPayments="undefined"
  wallet={[Function Object]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button
          text="Cancel"
        >
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Cancel
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <h2
      class="ion-text-center"
    >
      Wallet
    </h2>
    <ion-list>
      <ion-item
        class="wallet-item item"
      >
        <ion-avatar
          slot="start"
        >
          <span>
            <img
              alt="undefined"
              src="/assets/csaa-mobile/vectors/Debit.svg"
            />
          </span>
        </ion-avatar>
        <ion-label>
          <h3>
            test
          </h3>
          <p>
             1111 Debit 
          </p>
          <p>
             12345 Checking 
          </p>
        </ion-label>
        <ion-button
          fill="none"
          slot="end"
        >
           EDIT 
        </ion-button>
      </ion-item>
    </ion-list>
    <h2
      class="ion-text-center"
    >
      Payment Method
    </h2>
    <ion-list
      class="ion-margin-bottom"
    >
      <ion-item
        detail=""
        tabindex="0"
      >
        Credit or Debit Card
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Checking Account
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Savings Account
      </ion-item>
    </ion-list>
  </ion-content>
</app-csaa-payment-select-method>
`;

exports[`AutopaySelectMethodPage should display the Savings Account method 1`] = `
<app-csaa-payment-select-method
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  autopayLabel={[Function Function]}
  configService={[Function ConfigMockService]}
  currentTheme=""
  makePaymentService={[Function MakePaymentService]}
  newPaymentAccountMenu={[Function Array]}
  policy={[Function Object]}
  policyNumber={[Function String]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  tearDown$={[Function Subject]}
  upcomingPayments="undefined"
  wallet={[Function Object]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button
          text="Cancel"
        >
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Cancel
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <h2
      class="ion-text-center"
    >
      Wallet
    </h2>
    <ion-list>
      <ion-item
        class="wallet-item item"
      >
        <ion-avatar
          slot="start"
        >
          <span>
            <img
              alt="undefined"
              src="/assets/csaa-mobile/vectors/Debit.svg"
            />
          </span>
        </ion-avatar>
        <ion-label>
          <h3>
            test
          </h3>
          <p>
             1111 Debit 
          </p>
          <p>
             12345 Checking 
          </p>
        </ion-label>
        <ion-button
          fill="none"
          slot="end"
        >
           EDIT 
        </ion-button>
      </ion-item>
    </ion-list>
    <h2
      class="ion-text-center"
    >
      Payment Method
    </h2>
    <ion-list
      class="ion-margin-bottom"
    >
      <ion-item
        detail=""
        tabindex="0"
      >
        Credit or Debit Card
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Checking Account
      </ion-item>
      <ion-item
        detail=""
        tabindex="0"
      >
        Savings Account
      </ion-item>
    </ion-list>
  </ion-content>
</app-csaa-payment-select-method>
`;
